From 40e395e0991d2a77c7e33399f5de6f4ef9327dea Mon Sep 17 00:00:00 2001
From: Noam Camus <ncamus@habana.ai>
Date: Sat, 12 Oct 2019 18:50:50 +0300
Subject: [PATCH] [SW-0] gaudi: pcie: skip preboot_init_pcie() at mmNIC2_HBM_PLL_BASE WA

As part of HW erratum WA we reset asic while we clear PCIe enable strap
In such case do not do this in preboot and use dedicated FLOP to
identify this scenario

Change-Id: I45f820ae112ddeb2758ed7997b46c3bb89721cf8
---

diff --git a/gaudi/common.c b/gaudi/common.c
index 017cf91..6463c75 100644
--- a/gaudi/common.c
+++ b/gaudi/common.c
@@ -113,6 +113,10 @@
 	if (reset_cause == LKD_HARD_RESET_MAGIC)
 		return;
 
+	/* Skip if this is part of mmNIC2_HBM_PLL_BASE workaround */
+	if (regmap_readl(mmPSOC_GLOBAL_CONF_COLD_RST_FLOPS_2) == 1)
+		return;
+
 	preboot_init_pcie();
 }
 
